lib/api/src/ticketAPI.cpp

```diff
@@ -83,336 +94,587 @@ namespace irods::administration::ticket
+                       commandModifier2,
+                       commandModifier3,
+                       commandModifier4);
     }
-    int set_write_file_restriction(RxComm& conn, int ticket_ID, int numUses)
+    void remove_ticket_constraint(RxComm& conn, constraint::constraint& constraints)
     {
-        return ticket_manager(conn, "mod", std::to_string(ticket_ID), "write-file", std::to_string(numUses), "", "");
+        std::string_view command = modify_command;
+        std::string_view commandModifier1 = "remove";
+        std::string_view commandModifier2 = constraints.get_group_type();
+        std::string_view commandModifier3 = constraints.get_value();
+        std::string_view commandModifier4 = "";
+
+        ticket_manager(conn,
+                       command,
+                       constraints.get_ticket_identifier(),
+                       commandModifier1,
+                       commandModifier2,
+                       commandModifier3,
+                       commandModifier4);
     }
 
-    int set_write_byte_restriction(RxComm& conn, std::string_view ticket_name, int numUses)
+    void add_ticket_constraint(admin_tag, RxComm& conn, constraint::constraint& constraints)
     {
-        return ticket_manager(conn, "mod", ticket_name, "write-byte", std::to_string(numUses), "", "");
+        std::string_view command = modify_command;
+        std::string_view commandModifier1 = "add";
+        std::string_view commandModifier2 = constraints.get_group_type();
+        std::string_view commandModifier3 = constraints.get_value();
+        std::string_view commandModifier4 = "";
+
+        ticket_manager(conn,
+                       command,
+                       constraints.get_ticket_identifier(),
+                       commandModifier1,
+                       commandModifier2,
+                       commandModifier3,
+                       commandModifier4,
+                       true);
     }
-    int set_write_byte_restriction(RxComm& conn, int ticket_ID, int numUses)
+    void remove_ticket_constraint(admin_tag, RxComm& conn, constraint::constraint& constraints)
     {
-        return ticket_manager(conn, "mod", std::to_string(ticket_ID), "write-byte", std::to_string(numUses), "", "");
+        std::string_view command = modify_command;
+        std::string_view commandModifier1 = "remove";
+        std::string_view commandModifier2 = constraints.get_group_type();
+        std::string_view commandModifier3 = constraints.get_value();
+        std::string_view commandModifier4 = "";
+
+        ticket_manager(conn,
+                       command,
+                       constraints.get_ticket_identifier(),
+                       commandModifier1,
+                       commandModifier2,
+                       commandModifier3,
+                       commandModifier4,
+                       true);
     }
 
     int add_user(RxComm& conn, std::string_view ticket_name, std::string_view user)
```

Comment:
Don't forget to remove these functions since they are commented out in the header file.